### ISSUE TITLE
Controller manual mode

### ISSUE THREAD

**Issue by feelan03** [digitalocean/digitalocean-cloud-controller-manager#484]
It is not clear to me why the cloud controller works this way. I want to have 1 loadbalancer, not a new one for each service. Maybe it's worth making a choice of working cloud-contorller? For example, the sync option that works now (autosync) and the manual option that will receive only IP for the service, and the settings of the provider itself will be done using terraform. I think this kind of work makes sense.

**Comment by feelan03** [comment]
There is one more idea. If the name of the load balancer is specified, then work is done directly with it. At the moment, if a new service is created, it overwrites the data. I don't think this is the correct way for the controller to work.

**Comment by timoreimann** [comment]
The "one LB per Service" mapping is how CCM is designed upstream -- to my understanding, we do not and cannot change that approach without effectively diverging from the core upstream logic. Essentially, each CCM implementation is a set of hooks that follow a predefined workflow.

Your request isn't uncommon though. One recommendation to achieve that could be to use Ingress and demultiplex requests at the proxy level.

I'm not sure I understand your point about the LB name: it's another property of an LB that can be defined at any time through an annotation and causes the current LB configuration to be adjusted accordingly. I don't see why that would be out of line of how controllers should work -- let me know though if you feel like I'm missing something.

**Comment by feelan03** [comment]
Let's discuss. We have a Load Balancer called k8s-loadbalancer on which some settings are already hung. I want to connect a service to a specific loadbalacer by specifying the loadbalacer name while doing so so that the settings are saved.
I do it like this:
```
service:
   annotations:
     service.beta.kubernetes.io/do-loadbalancer-name: k8s-loadbalancer
     service.beta.kubernetes.io/do-loadbalancer-hostname: "grafana.do.dbservice.tech"
   type: LoadBalancer
   port: 3000
```
Expected behavior - the created loadbalacer will not be deleted, but will be modified. The entry_port & target_port are taken from the k8s services and these values are added to the loadbalancer digital ocean using the API

**Comment by timoreimann** [comment]
@feelan03 so you are saying that  in your scenario, you see an existing load balancer getting deleted when you specify the name annotation? That should certainly not happen, and instead what you described should be carried out (existing load balancer should be modified/renamed).

Could you provide a series of reproduction steps to confirm the behavior and allow analyzing the issue further?

**Comment by timoreimann** [comment]
@feelan03 the likely reason the that VictoraMetrics controller(?) is creating a new LB is that the service annotation referencing the LB ID is missing within the `serviceSpec` so CCM has no way to associate the LB unambiguously. See also [the docs we have in this regard](https://github.com/digitalocean/digitalocean-cloud-controller-manager/blob/462fe5bad45a5bd4925bdd1e52792bfbdf8c0686/docs/getting-started.md#load-balancer-id-annotations).

**Comment by feelan03** [comment]
@timoreimann It's very strange anyway. If you look at the yaml for this service, you can see the difference kubernetes.digitalocean.com/load-balancer-id. How could this happen, why didn't he specify an ID that already exists. Don't you think this is strange? Let me know if I'm doing something wrong, maybe I'm missing something.
![ScreenShot-2022-07-13](https://user-images.githubusercontent.com/57330758/178705561-7e35c682-c4c2-4c72-aa66-523e6dc285d3.png)
![ScreenShot-2022-07-13](https://user-images.githubusercontent.com/57330758/178705625-a9f6c0d6-3f24-4388-8802-1a1c7257f9c1.png)

**Comment by feelan03** [comment]
But what if you try to force the id load balancer.  maybe then he can earn?

**Comment by timoreimann** [comment]
The load balancer ID is only added to the Service object _after_ the LB create request has returned (i.e., when a Service object of type `LoadBalancer` is reconciled by CCM for the first time). If the ID is missing (and CCM cannot derive it from the default, auto-generated LB name), then CCM will assume that a new LB must be created.

In your VictoriaMetrics manifest, you haven't provided the LB ID annotation so CCM cannot relate it to the existing LB. I think that if you include and copy the annotation from an existing Service / LB, then it could work. (No guarantees though since I don't know how that controller works in detail.)

**Comment by feelan03** [comment]
Strange, I specified the name Load Balancer and type Load Balancer. For me, it's still surprising that CCM did not find this LB. I will definitely test the manifesto with the settings from the finished LB and the ladies feedback. Thank you.

**Comment by timoreimann** [comment]
I just confirmed in the code that, without an LB ID annotation available, CCM should still try to discover a pre-existing LB by looking up the custom and default (auto-generated) name. Given that and the fact that you specified the custom name annotation on the VictoriaMetrics manifest, I'd have expected an existing LB to be found.

I am more and more thinking that it's somehow related to the VM controller behavior. Might be worth for you to look up the actual implementation (and perhaps any logs) to see what it does exactly.

**Comment by timoreimann** [comment]
Sorry I don't speak what appears to be Russian. 🙂

**Comment by feelan03** [comment]
I'm sorry, I forgot to translate into English

**Comment by feelan03** [comment]
Now I'm seeing different behavior.
I removed absolutely all loadbalancer in DO. Removed all services that used them. And I decided to do it all over again, so that there would be no problems with the CCM.
1. I installed ingress-controller
`helm upgrade --install ingress-nginx ingress-nginx/ingress-nginx --create-namespace --namespace=ingress-nginx`
When installing ingress-controler it created the loadbalancer itself
2. I copied the ID of the created loadbalancer to myself. Took it with `k edit svc ingress-nginx-controller -n ingress-nginx -o yaml`
3. I edited the values ​​of helm chart grafana. This is how he looks now
```
service:
  annotations:
    kubernetes.digitalocean.com/load-balancer-id: f7aaf1a4-6679-49ca-aa1a-c80708812d5c
    service.beta.kubernetes.io/do-loadbalancer-hostname: "grafana.do.dbservice.tech"
  type: LoadBalancer
  port: 3000
```
4. Launched helm chart and started observing the loadbalancer settings. Action expected: adding port 3000 to the created load balancer. Reality: Deleted settings 80:30194/TCP,443:30490/TCP and created setting
3000:30334/TCP

Could this be related to digital-ocean-manager being in separate namespaces?
